### DISCUSSION TITLE
Livewire.dispatch and $wire.dispatch or $wire.dispatchSelf from javascript, they don't work for me

### DISCUSSION THREAD

**Discussion by Rusbelito** [livewire/livewire#7795] [1. Help]
I'm trying to listen to an event launched from JavaScript but I can't do it:

I am doing this, following the documentation at:
https://livewire.laravel.com/docs/javascript#interacting-with-events
and in :
https://livewire.laravel.com/docs/events#dispatching-events-from-component-scripts
but neither of them works for me, I'm trying something similar to this:

```
<div class="flex w-full h-full items-center justify-center text-gray-300 px-16">
    {{$dispatch}}
</div>

@script
    <script>
    $wire.dispatch('post-created');
    Livewire.dispatch('post-created');
    </script>
@endscript
```
and listen to it like this:

```
#[On('post-created')]
public function post(){
    $this->$dispatch = 'yes' ;
}

```

I'm also trying to add the launchers.: `document.addEventListener('livewire:init', () => {...}) `

but it doesn't work, I sometimes change the name of my function and make sure that I was calling On and if I dispatch from the same component the listener works but when I do it from javascript it doesn't, am I ignoring something? 

here an example: https://wirebox.app/b/x517w

**Comment by celvintr** [comment]
tengo el mismo problema, pudistes resolver?

**Comment by danie-ramdhani** [comment]
you need to wrap it with `$nextTick`:

```php
#[\Livewire\Attributes\On('post-created')]
public function create()
{
    dd('created!');
}
```

```blade
@script
  <script>
      $nextTick(() => {
          $wire.$dispatch('post-created');
      });
  </script>
@endscript
```

**Comment by calebporzio** [comment]
This has been fixed here: https://github.com/livewire/livewire/pull/8226

It will be available in the release next week.
